module_utils/network_lsr/argument_validator.py

```diff
@@ -531,6 +533,13 @@ def _validate_post(self, value, name, result):
                 raise ValidationError(
                     name, "'dhcp4_send_hostname' is only valid if 'dhcp4' is enabled"
                 )
+        if result["ipv6_disabled"] is True:
+            result["auto6"] = None
```

Comment:
if "ipv6_disable" is on, then you should check that the user didn't set `auto6:yes` and didn't configure any IPv6 addresses. That would be conflicting and should be rejected.

Also, a major point of `ArgValidator` is to *normalize* the values. In yaml, a setting like `ipv6_disabled` has three possible values: `True`, `False` and unspecified (`None`). After `_validate_post()`, `None` should be normalized to either True or False.

Reply:
like you mentioned in IRC,  if we don't specify any methods, then "auto" will be enabled by default (as ArgValidator already normalizes). then conflicts/error happened everytime when we set "ipv6_disable" on.

Reply:
this sets `result["auto6"] = None`. That cannot be right because `_validate_post()` should normalize `auto6` to a boolean and not leave it undecided. It is only left undecided during `_validate_impl()`, so we can distinguish here between "unset" and explicitly set.

Reply:
You said "After _validate_post(), None should be normalized to either True or False.", then `auto6` is either true or false , `ipv6_disabled` is either true or false. I did not see how we can achieve  `'auto6' and 'ipv6_disabled' are mutually exclusive`.

Reply:
Is it clear now? I think the code is now good, marking this question as "resolved". Please unresolve is something is wrong.